internal/resources/managementclusterregistration/resource_management_cluster_registration.go

```diff
@@ -0,0 +1,193 @@
+/*
+Copyright © 2023 VMware, Inc. All Rights Reserved.
+SPDX-License-Identifier: MPL-2.0
+*/
+
+package managementclusterregistration
+
+import (
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+
+	clustermodel "github.com/vmware/terraform-provider-tanzu-mission-control/internal/models/cluster"
+	managementclusterregistrationmodel "github.com/vmware/terraform-provider-tanzu-mission-control/internal/models/managementclusterregistration"
+	"github.com/vmware/terraform-provider-tanzu-mission-control/internal/resources/common"
+)
+
+func ResourceManagementClusterRegistration() *schema.Resource {
+	return &schema.Resource{
+		Schema: managementClusterRegistrationSchema,
+		//CreateContext: , TODO
+		//ReadContext: , TODO
+		//UpdateContext: , TODO
+		//DeleteContext: , TODO
+		Description: "Tanzu Mission Control Management Cluster Registration Resource",
+	}
+}
+
+var managementClusterRegistrationSchema = map[string]*schema.Schema{
+	NameKey: {
+		Type:        schema.TypeString,
+		Description: "Name of this management cluster",
+		Required:    true,
+		ForceNew:    true,
+	},
+	common.MetaKey:     common.Meta,
+	specKey:            managementClusterRegistrationSpecSchema,
+	registerClusterKey: registerCluster,
+	StatusKey: {
+		Type:        schema.TypeMap,
+		Description: "Status of the cluster",
+		Computed:    true,
+		Elem:        &schema.Schema{Type: schema.TypeString},
+	},
+	waitKey: {
+		Type:        schema.TypeString,
+		Description: "Wait timeout duration.",
+		Default:     "default",
+		Optional:    true,
+	},
+}
+
+var registerCluster = &schema.Schema{
+	Type:     schema.TypeList,
+	Optional: true,
+	MinItems: 1,
+	MaxItems: 1,
+	Elem: &schema.Resource{
+		Schema: map[string]*schema.Schema{
+			registerClusterKubeConfigPathKey: {
+				Type:        schema.TypeString,
+				Description: "Register cluster KUBECONFIG path",
+				ForceNew:    true,
+				Optional:    true,
+			},
+			registerClusterKubeConfigRawKey: {
+				Type:        schema.TypeString,
+				Description: "Register cluster KUBECONFIG",
+				Optional:    true,
+				ForceNew:    true,
+				Sensitive:   true,
+			},
+			registerClusterDescriptionKey: {
+				Type:         schema.TypeString,
+				Description:  "Register cluster description",
+				Optional:     true,
+				ValidateFunc: validation.StringIsNotWhiteSpace,
+			},
+		},
+	},
+}
+var KubeConfigWayAllowed = [...]string{registerClusterKubeConfigPathKey, registerClusterKubeConfigRawKey}
+
+var managementClusterRegistrationSpecSchema = &schema.Schema{
```

Comment:
nit: we can have all the vars defined in a single block

Reply:
Yes we can.